ui/selectors/multichain.ts

```diff
@@ -111,17 +123,55 @@ export function getMultichainNetwork(
 ): MultichainNetwork {
   const isEvm = getMultichainIsEvm(state, account);
 
-  // EVM networks
-  const evmNetworks: ProviderConfig[] = getAllNetworks(state);
-  const evmChainId: ChainId = getCurrentChainId(state);
-
   if (isEvm) {
-    const evmNetwork: ProviderConfig =
-      evmNetworks.find((provider) => provider.chainId === evmChainId) ??
-      getProviderConfig(state); // We fallback to the original selector otherwise
+    // EVM networks
+    const evmChainId: string = getCurrentChainId(state);
+
+    // TODO: Update to use network configurations when @metamask/network-controller is updated to 20.0.0
+    // ProviderConfig will be deprecated to use NetworkConfigurations
+    // When a user updates a network name its only updated in the NetworkConfigurations.
+    const evmNetwork: ProviderConfigWithImageUrlAndExplorerUrl =
+      getProviderConfig(state);
+    // Fallback to a known network image if network configuration does not defined it
+    const evmChainIdKey =
+      evmChainId as keyof typeof CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP;
+    if (
+      !evmNetwork?.rpcPrefs?.imageUrl &&
+      evmChainIdKey in CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP
+    ) {
+      evmNetwork.rpcPrefs = {
+        ...evmNetwork.rpcPrefs,
+        imageUrl: CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP[evmChainIdKey],
+      };
+    }
+
+    let nickname;
+    if (evmNetwork.type === NETWORK_TYPES.RPC) {
+      // Could be undefined for default configurations.
+      const evmNetworkConfigurations = getNetworkConfigurations(state);
+      const evmNetworkConfiguration =
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
I'm not sure if this is true though? It's definitely true for the `rpcUrl` but IDK about `id`, do you know?

```suggestion
        // id will always be defined for custom networks.
        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```